src/ui_txn.c

```diff
@@ -42,6 +45,22 @@ all_zero_key(uint8_t *buf)
   return 1;
 }
 
+static char *
+b64hash_data(unsigned char *data, size_t data_len)
+{
+  static char b64hash[45];
+  static unsigned char hash[32];
```

Comment:
`hash[32]` can stay on the stack, since we don't return pointers to it, and making it `static` decicates some SRAM for it at all times.